### PR TITLE
feat(electron): use affine native

### PR DESCRIPTION
**Pull request by Brooooooklyn**

### CODE REVIEW

File: apps/electron/layers/main/src/handlers/db/ensure-db.ts [hunk 1]
```diff
@@ -1,94 +1,130 @@
...
+import { ts } from '../../utils';
 import type { WorkspaceSQLiteDB } from './sqlite';
 import { openWorkspaceDatabase } from './sqlite';
 
-const dbMapping = new Map<string, Promise<WorkspaceSQLiteDB>>();
-const dbWatchers = new Map<string, () => void>();
+const databaseInput$ = new Subject<string>();
+const terminate$ = app ? fromEvent(app, 'before-quit') : NEVER;
+
+const database$ = databaseInput$.pipe(
```
Comment (pengx17) [inline apps/electron/layers/main/src/handlers/db/ensure-db.ts:38]: I think we may still need to have mutliple db's to be watched at the same time soon later, e.g. we have multiple-tab/window and each of them need to watch for db file changes.

Comment (Brooooooklyn) [reply apps/electron/layers/main/src/handlers/db/ensure-db.ts:38]: The watch logics are not changed at all; the `database$` emits different `workspaceId` from the upstream

File: apps/electron/layers/main/src/handlers/db/ensure-db.ts [hunk 2]
```diff
@@ -1,94 +1,130 @@
...
+  distinctUntilChanged(),
+  groupBy(identity),
+  mergeMap(workspaceDatabase$ =>
+    workspaceDatabase$.pipe(
+      // only open the first db, and emit it to the downstream
```
Comment (Brooooooklyn) [inline apps/electron/layers/main/src/handlers/db/ensure-db.ts:43]:
```suggestion
      // only open the first db with the same workspaceId, and emit it to the downstream
```

File: apps/electron/layers/main/src/handlers/db/ensure-db.ts (hunk 1 above)
Comment (Brooooooklyn) [reply apps/electron/layers/main/src/handlers/db/ensure-db.ts:38]: See my comment below

**Review by pengx17 — DISMISSED** [review]
flawless

**Review by Himself65 — APPROVED** [review]

**Review by pengx17 — APPROVED** [review]
LGTM

**Pull request merged**